docs/src/main/tut/kleisli.md

```diff
@@ -0,0 +1,237 @@
+
+An example of a `Monad` instance for `Kleisli` would be:
+
+```tut
+// We can define a FlatMap instance for Kleisli if the F[_] we chose has a FlatMap instance
+// Note the input type and F are fixed, with the output type left free
+implicit def kleisliFlatMap[F[_], Z](implicit F: FlatMap[F]): FlatMap[Kleisli[F, Z, ?]] =
+  new FlatMap[Kleisli[F, Z, ?]] {
+    def flatMap[A, B](fa: Kleisli[F, Z, A])(f: A => Kleisli[F, Z, B]): Kleisli[F, Z, B] =
+      Kleisli(z => F.flatMap(fa.run(z))(a => f(a).run(z)))
+
+    def map[A, B](fa: Kleisli[F, Z, A])(f: A => B): Kleisli[F, Z, B] =
+      Kleisli(z => F.map(fa.run(z))(f))
+  }
+```
+
+## Other uses
+### Monad Transformer
+Many data types have a monad transformer equivalent that allows us to compose the `Monad` instance of the data
+type with any other `Monad` instance. For instance, `OptionT[F[_], A]` allows us to compose the monadic properties
+of `Option` with any other `F[_]`, such as a `List`. This allows us to work with nested contexts/effects in a
+nice way, say, in for comprehensions.
+
+`Kleisli` can be viewed as the monad transformer for functions. Recall that at its essence, `Kleisli[F, A, B]`
+is just a function `A => F[B]`, with niceties to make working with the value we actually care about, the `B`, easy.
+`Kleisli` allows us to take the effects of functions and have them play nice with the effects of any other `F[_]`.
+
+This may raise the question, what exactly is the "effect" of a function?
+
+Well, if we take a look at any function, we can see it takes some input and produces some output with it, without
+having touched the input (assuming a functional environment). That is, we take a read-only value, and produce some
+value with it. For this reason, the type class instances for functions often refer to the function as a `Reader`.
+For instance, it is common to hear about the `Reader` monad. In the same spirit, Cats defines a `Reader` type alias
+along the lines of:
+
+```tut
+// We want A => B, but Kleisli provides A => F[B]. To make the types/shapes match,
+// we need an F[_] such that providing it a type A is equivalent to A
+// This can be thought of as the type-level equivalent of the identity function
+type Id[A] = A
+
+type Reader[A, B] = Kleisli[Id, A, B]
+object Reader {
+  // Lifts a plain function A => B into a Kleisli, giving us access
+  // to all the useful methods and type class instances
+  def apply[A, B](f: A => B): Reader[A, B] = Kleisli[Id, A, B](f)
+}
+
+type ReaderT[F[_], A, B] = Kleisli[F, A, B]
+val ReaderT = Kleisli
+```
+
+The `ReaderT` value alias exists to allow users to use the `Kleisli` companion object as if it were `ReaderT`, if
+they were so inclined.
+
+The topic of functions as a read-only environment brings us to our next common use case of `Kleisli` - configuration.
+
+### Configuration
+In functional programming it is often advocated to build many small modules, and then compose them together to
```

Comment:
Maybe this can be simplified to "Functional programming advocates the creation of programs and modules by composing smaller, simpler modules." ?